src/devices/proformtreadmill/proformtreadmill.cpp

```diff
@@ -2062,6 +2155,18 @@ void proformtreadmill::characteristicChanged(const QLowEnergyCharacteristic &cha
 
     lastPacket = newValue;
 
+    if (proform_treadmill_705_cst && currentSpeed().value() == 0)
+    {
+        QString hexString = newValue.toHex(' ');
+        if (hexString.startsWith("fe 02 25 04 00 00") && newValue.at(17) == 0x01)
+        {
+            emit debug(QStringLiteral("start button pressed on console!"));
+            Distance = 0;
```

Comment:
@Novamusix i don't like that Distance = 0, the distance is resetted on qz when you press the commands on the qz top bar.

Reply:
When running via an Android/iPhone, this may be the case. However, when running headless, the following scenario is encountered:

**QZ (RPi) connected to treadmill:**
- Start a new session via a connected app (e.g., Runna).
- Treadmill metrics progress on the console.
**Pause:**
- Press stop on the console once to pause the session
- Treadmill speed is zero.
- Press start again to resume.
- No start Bluetooth sequence is sent.
**Finish:**
- Press stop once to pause and stop again to finish the session.
- Pressing start will not do anything
- No start Bluetooth sequence is sent
**Reset:**
- Press stop once to pause, a second time to finish, and a third time to reset the treadmill session. 
- The console now displays zero values. 
- Pressing start will now send a Bluetooth sequence.
- However, because the distance is calculated in QZ and not retrieved from the treadmill, it will keep adding to the previous sessions distance.

Knowing that the start Bluetooth sequence is only sent when first powered on or when having been reset with the console showing zero values, meant resetting the distance value ensured it matches what it is shown on the treadmill.

And that when sending a speed request from an app, we can initiate start sequence which is needed to unlock the treadmill. The risk being without checking distance is zero the initiate sequence could be sent when paused.

There may be better ways of handling this, but this approach seem to work for the above scenarios.

Reply:
ok it makes sense. i conditioned this only if the homeform is not available, this means a headless session. Could you try this branch and check if it's still working fine on your end?